<?xml version="1.0"?>
<launch>
    <arg name="debug" default="true" />
    <arg name="face_feature_topic" default="/migrave_perception/openface_ros/faces" />
    <arg name="face_feature_topic_ext_up" default="/migrave_perception/openface_ros_ext_up/faces" />
    <arg name="face_feature_topic_ext_down" default="/migrave_perception/openface_ros_ext_down/faces" />

    <arg name="of_debug_img_topic" default="/migrave_perception/openface_ros/debug_image" />
    <arg name="of_debug_img_topic_ext_up" default="/migrave_perception/openface_ros_ext_up/debug_image" />
    <arg name="of_debug_img_topic_ext_down" default="/migrave_perception/openface_ros_ext_down/debug_image" />

    <arg name="overall_engagement_topic" default="/migrave_perception/person_state_estimator/affective_state" />
    <arg name="camera_names" default="[rgbd, ext_up, ext_down]" />
    <arg name="engagement_topics" default="[/migrave_perception/person_state_estimator_rgbd/affective_state,/migrave_perception/person_state_estimator_ext_up/affective_state,/migrave_perception/person_state_estimator_ext_down/affective_state]" />
    <arg name="avg_engagement_estimation_server_name" default="/migrave_perception/get_avg_engagement" />
    <arg name="engagement_cache_s" default="5.0" />

    <node pkg="migrave_person_state_estimation" type="person_state_estimator" name="person_state_estimator_rgbd" output="screen"
     ns="migrave_perception" >
        <param name="config_path" type="str" value="$(find migrave_person_state_estimation)/ros/config/person_state_estimation_config.yaml" />
        <param name="face_feature_topic" type="str" value="$(arg face_feature_topic)" />
        <param name="audio_feature_topic" type="str" value="/audio_features" />
        <param name="skeleton_topic" type="str" value="/skeletons" />
        <param name="game_performance_topic" type="str" value="/game_performance" />
        <param name="debug" type="bool" value="$(arg debug)" />
        <param name="of_debug_img_topic" type="str" value="$(arg of_debug_img_topic)" />
        <param name="camera_name" type="str" value="rgbd" />
    </node>

    <node pkg="migrave_person_state_estimation" type="person_state_estimator" name="person_state_estimator_ext_up" output="screen"
     ns="migrave_perception" >
        <param name="config_path" type="str" value="$(find migrave_person_state_estimation)/ros/config/person_state_estimation_config_up.yaml" />
        <param name="face_feature_topic" type="str" value="$(arg face_feature_topic_ext_up)" />
        <param name="audio_feature_topic" type="str" value="/audio_features" />
        <param name="skeleton_topic" type="str" value="/skeletons" />
        <param name="game_performance_topic" type="str" value="/game_performance" />
        <param name="debug" type="bool" value="$(arg debug)" />
        <param name="of_debug_img_topic" type="str" value="$(arg of_debug_img_topic_ext_up)" />
        <param name="camera_name" type="str" value="ext_up" />
    </node>

    <node pkg="migrave_person_state_estimation" type="person_state_estimator" name="person_state_estimator_ext_down" output="screen"
     ns="migrave_perception" >
        <param name="config_path" type="str" value="$(find migrave_person_state_estimation)/ros/config/person_state_estimation_config_down.yaml" />
        <param name="face_feature_topic" type="str" value="$(arg face_feature_topic_ext_down)" />
        <param name="audio_feature_topic" type="str" value="/audio_features" />
        <param name="skeleton_topic" type="str" value="/skeletons" />
        <param name="game_performance_topic" type="str" value="/game_performance" />
        <param name="debug" type="bool" value="$(arg debug)" />
        <param name="of_debug_img_topic" type="str" value="$(arg of_debug_img_topic_ext_down)" />
        <param name="camera_name" type="str" value="ext_down" />
    </node>

    <node pkg="migrave_person_state_estimation" type="person_state_merger" name="person_state_merger" output="screen" ns="migrave_perception">
        <rosparam param="camera_names" subst_value="True">$(arg camera_names)</rosparam>
        <rosparam param="engagement_topics" subst_value="True">$(arg engagement_topics)</rosparam>
        <param name="overall_engagement_topic" type="str" value="$(arg overall_engagement_topic)" />
        <param name="avg_engagement_estimation_server_name" type="str" value="$(arg avg_engagement_estimation_server_name)" />
        <param name="engagement_cache_s" type="double" value="$(arg engagement_cache_s)" />
    </node>
</launch>
